### ISSUE TITLE
Use gunicorn for Inventory CKAN

### ISSUE THREAD

**Issue by adborden** [GSA/data.gov#1383]
## User Story

As a Data.gov developer, I want CKAN using gunicorn so that it behaves more like a process as per [12factor.net](https://12factor.net) and reduces the dependence on apache2.

## Acceptance Criteria
- [x] WHEN Inventory is deployed the web app is served as a gunicorn process managed by supervisor SO it is restarted if crashed and logs are tracked appropriately.

**Comment by adborden** [comment]
For a long time, we've shipped a custom python version (2.7.10 for Trusty, 2.7.16 for Bionic) on the platform. We download a "recent" version of Python, compile and install it. This causes some problems, most we've skirted around. The biggest is that for apache mod_wsgi we set [WSGIPythonHome](https://modwsgi.readthedocs.io/en/develop/configuration-directives/WSGIPythonHome.html) to our custom python install but the docs say:

> Note that the Python installation being referred to using this directive must be the same major/minor version of Python that mod_wsgi was compiled for. If you want to use a different version of major/minor version of Python than currently used, you must recompile mod_wsgi against the alternate version of Python.

Trusty ships with Python 2.6, so the OS provided mod_wsgi shouldn't really be configured with our version of Python 2.7 but somehow it still works (or at least appears to work). Managing CKAN in a separate process gives us more control, we can specify whatever version of Python and use an isolated virtualenv.

Bionic ships with python 2.7, so in theory there shouldn't be an issue, **but** it's possible our version of 2.7 conflicts with the system version. For example, when trying to configure mod_wsgi with the system version, I saw this in /var/log/apache2/error.log:
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/site.py", line 554, in <module>
    main()
  File "/usr/lib/python2.7/site.py", line 536, in main
    known_paths = addusersitepackages(known_paths)
  File "/usr/lib/python2.7/site.py", line 272, in addusersitepackages
    user_site = getusersitepackages()
  File "/usr/lib/python2.7/site.py", line 247, in getusersitepackages
    user_base = getuserbase() # this will also set USER_BASE
  File "/usr/lib/python2.7/site.py", line 237, in getuserbase
    USER_BASE = get_config_var('userbase')
  File "/usr/lib/python2.7/sysconfig.py", line 587, in get_config_var
    return get_config_vars().get(name)
  File "/usr/lib/python2.7/sysconfig.py", line 533, in get_config_vars
    _init_posix(_CONFIG_VARS)
  File "/usr/lib/python2.7/sysconfig.py", line 417, in _init_posix
    from _sysconfigdata import build_time_vars
  File "/usr/lib/python2.7/_sysconfigdata.py", line 6, in <module>
    from _sysconfigdata_nd import *
ImportError: No module named _sysconfigdata_nd
```

This says to me mod_wsgi is initializing, but getting hung up on internal python imports not existing.

**Comment by adborden** [comment]
If we were to implement, there is prior work in pycsw. The [datagov-deploy-pyscw](https://github.com/GSA/datagov-deploy-pycsw) role installs pycsw to an isolated application user, with its own pyenv python install.

The pycsw [app itself](https://github.com/GSA/pycsw) includes gunicorn in its requirements file, and the wsgi script is committed to the application, not the ansible role.
